### PR TITLE
1911 - Fixed Patient_age display negative

### PR DESCRIPTION
**Pull request by oslynn**
This PR is to fix the patient's age bug which give negative value.  In addition to fix the bug, it also added the unit test 
for patient's age validation to make sure it is a good patient age to persist.

Test Steps:
1. Shutdown docker-compose
`docker-compose` down
2. Remove the prime-router_prime_dev container from the docker
`docker` container rm <prime-router_primer_dev container ID>
3.) Clone the repository and build prime-router_prime_dev docker image 
./gradlew clean package
3.) Start the docker-compose
docker-compose up.
4.) Run the following command to persist the report to the covid_result_metadata database table.
curl -X POST -H "client:simple_report" -H "content-type:text/csv" --data-binary "@./junk/PatientAgeTest.csv" "http://localhost:7071/api/reports"
4.) Note the report_id ("id") from the response resource.
5.) In Postgres Data users can query the table using the following:
SELECT patient_age from covid_result_metadata where report_id= '<noted id in step 4 above>'
user will NOT see the patient_age: Negative Number.

Test file: 
[PatientAgeTest.csv](https://github.com/CDCgov/prime-reportstream/files/7497898/PatientAgeTest.csv)

Screenshot:
image

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [X] Tested locally?
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Added tests?
   ./gradlew test

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
- #issue - List GitHub issues this PR fixes
- Fixed patient_age display negative number ticket: https://app.zenhub.com/workspaces/prime-reportstream-5ff4833beb3e08001a4cacae/issues/cdcgov/prime-reportstream/1911

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/Report.kt [hunk 1]
```diff
@@ -558,7 +558,9 @@ class Report : Logging {
                     it.patientAge = row.getStringOrNull("patient_dob").let { dob ->
                         try {
                             val d = LocalDate.parse(dob, Element.dateFormatter)
-                            if (d != null && it.specimenCollectionDateTime != null) {
+                            if (d != null && it.specimenCollectionDateTime != null &&
+                                (d.isBefore(it.specimenCollectionDateTime))
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/Report.kt:562]: Add a quick comment just saying why you are using DOB to check the date.

File: prime-router/src/test/kotlin/ReportTests.kt [hunk 2]
```diff
@@ -150,6 +152,40 @@ class ReportTests {
...
+            Element("message_id"), Element("specimen_collection_date_time"), Element("patient_dob")))
+
+        /**
+         * Add Rows values to the table
+         */
+        val oneReport = Report(schema = one, values = listOf(
+            listOf("0", "202110300809", "30300102"),        // Invalid collection datetime, bad DOB -> patient_age=null
+            listOf("1", "202110300809-0501", "30300101"),   // Good collection datetime, Invalid DOB -> patient_age=null
+            listOf("2", "202110300809", "20190101"),        // Invalid collection datetime -> patient_age = null
+            listOf("3", "adfadf", "!@!*@(7"),               // Garbage -> patient_age = null
```
Comment (carlosfelix2) [inline prime-router/src/test/kotlin/ReportTests.kt:171]: Nice!

**Review by carlosfelix2 — APPROVED** [review]
PR description needs to be populated correctly

**Review by jimduff-usds — CHANGES_REQUESTED** [review]
Does not meet acceptance criteria -- Needs to use patient_age if it exists.  This was repeated twice in the acceptance criteria of the ticket:

- As part of the fix, add a test to see if the incoming data has a patient_age value. If it does, just use it as-is. If not, then attempt to calculate using patient_dob.
- use the incoming patient_age is used, if it exists and is valid. (Is a number greater than zero)

**Review by jimduff-usds — APPROVED** [review]

**Review by clediggins-usds — APPROVED** [review]

**Pull request merged**